sprocs/sync_assessments.sql

```diff
@@ -20,88 +21,157 @@ DECLARE
     new_assessment_question_id bigint;
     new_assessment_question_ids bigint[];
 BEGIN
-    -- The outermost structure of the JSON blob is an array of assessments
-    FOR assessment IN SELECT * FROM JSONB_ARRAY_ELEMENTS(sync_assessments.assessments) LOOP
-        new_assessment_question_ids := array[]::bigint[];
+    -- Move all our data into a temporary table so it's easier to work with
+    DROP TABLE IF EXISTS disk_assessments;
+    CREATE TEMPORARY TABLE disk_assessments (
+        tid TEXT NOT NULL,
+        uuid uuid,
+        errors TEXT,
+        warnings TEXT,
+        data JSONB
+    ) ON COMMIT DROP;
+    INSERT INTO disk_assessments (
+        tid,
+        uuid,
+        errors,
+        warnings,
+        data
+    ) SELECT
+        entries->>0,
+        (entries->>1)::uuid,
+        entries->>2,
+        entries->>3,
+        (entries->4)::JSONB
+    FROM UNNEST(disk_assessments_data) AS entries;
```

Comment:
I think it's safer to use JSON objects rather than arrays with `jsonb_to_recordset()`, as described in the final two bullet points of the doc section: https://prairielearn.readthedocs.io/en/latest/dev-guide/#database-access

For an example, see: https://github.com/PrairieLearn/PrairieLearn/blob/1b249374d6f15e1468d53ce4e2b1a7efa02650cb/sprocs/sync_news_items.sql#L33